tutorials/Window_API_new/README.md

```diff
@@ -0,0 +1,492 @@
+
+    Window(
+        state,
+        title = "Counter",
+        onCloseRequest = { state.isVisible = false }
+    ) {
+        var counter by remember { mutableStateOf(0) }
+        LaunchedEffect(Unit) {
+            while (true) {
+                counter++
+                delay(1000)
+            }
+        }
+        Text(counter.toString())
+    }
+
+    if (!state.isVisible && state.isOpen) {
+        Tray(
+            remember { getTrayIcon() },
+            hint = "Counter",
+            onAction = { state.isVisible = true },
+            menu = {
+                Item("Exit", onClick = { state.isOpen = false })
+            },
+        )
+    }
+}
+
+fun getTrayIcon(): BufferedImage {
+    val size = 256
+    val image = BufferedImage(size, size, BufferedImage.TYPE_INT_ARGB)
+    val graphics = image.createGraphics()
+    graphics.color = Color.orange
+    graphics.fillOval(0, 0, size, size)
+    graphics.dispose()
+    return image
+}
+```
+![](hide_instead_of_close.gif)
+
+If application has multiple windows then it is better to hoist its state into a separate class and open/close window in response of some list state changes (see [notepad example](https://github.com/JetBrains/compose-jb/tree/master/examples/notepad) for more complex use cases):
+```kotlin
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.window.MenuBar
+import androidx.compose.ui.window.Window
+import androidx.compose.ui.window.application
+
+@OptIn(ExperimentalComposeUiApi::class)
+fun main() = application {
+    val applicationState = remember { MyApplicationState() }
+    for (window in applicationState.windows) {
+        MyWindow(window)
+    }
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun MyWindow(
+    state: MyWindowState
+) = Window(title = state.title, onCloseRequest = state::close) {
+    MenuBar {
+        Menu("File") {
+            Item("New window", onClick = state.openNewWindow)
+            Item("Exit", onClick = state.exit)
+        }
+    }
+}
+
+private class MyApplicationState {
+    val windows = mutableStateListOf<MyWindowState>()
+
+    init {
+        windows += MyWindowState("Initial window")
+    }
+
+    fun openNewWindow() {
+        windows += MyWindowState("Window ${windows.size}")
+    }
+
+    fun exit() {
+        windows.clear()
+    }
+
+    private fun MyWindowState(
+        title: String
+    ) = MyWindowState(
+        title,
+        openNewWindow = ::openNewWindow,
+        exit = ::exit,
+        windows::remove
+    )
+}
+
+private class MyWindowState(
+    val title: String,
+    val openNewWindow: () -> Unit,
+    val exit: () -> Unit,
+    private val close: (MyWindowState) -> Unit
+) {
+    fun close() = close(this)
+}
+```
+(don't forget to define a custom `onCloseRequest` to remove a window from the list)
```

Comment:
Is this comment still relevant? Shalln't it be in the example?

Reply:
I think we don't need it, removed.